program/rust/src/processor.rs

```diff
@@ -84,5 +103,29 @@ pub fn process_instruction(
         DelProduct => del_product(program_id, accounts, instruction_data),
         UpdPermissions => upd_permissions(program_id, accounts, instruction_data),
         SetMaxLatency => set_max_latency(program_id, accounts, instruction_data),
+        InitPriceFeedIndex => init_price_feed_index(program_id, accounts, instruction_data),
+    }
+}
+
+fn reserve_new_price_feed_index(permissions_account: &AccountInfo) -> Result<u32, ProgramError> {
```

Comment:
this feels like it should be in `permission.rs`

Reply:
`permission.rs` is a part of `accounts` module that doesn't really contain a lot of other code like this. I think it better fits in the `processor` module.